File: frontend/src/types/season.ts
/** Mirrors backend app/models/season.py. A season is one calendar year;
 * see backend/app/services/season_service.py for why it's derived from
 * session dates rather than stored. */
export interface Season {
  number: number
  year: number
  start_date: string
  end_date: string
  is_current: boolean
}

File: backend/app/models/season.py
from datetime import date

from pydantic import BaseModel


class Season(BaseModel):
    """One calendar year of club play. See app/services/season_service.py
    for why seasons are derived from session dates instead of stored."""

    number: int
    year: int
    start_date: date
    end_date: date
    is_current: bool
